packages/block-editor/README.md

```diff
@@ -558,6 +558,25 @@ _Related_
 
 Undocumented declaration.
 
+<a name="useBlockProps" href="#useBlockProps">#</a> **useBlockProps**
+
+This hook is used to lightly mark an element as a block element. Call this
+hook and pass the returned props to the element to mark as a block. If you
+define a ref for the element, it is important to pass the ref to this hook,
+which the hooks in turn will pass to the component through the props it
+returns. Optionally, you can also pass any other props through this hook, and
+they will be merged and returned.
```

Comment:
Reading the description, a few things stand out to me:

 - Should we state explicitly that it should be used on the outer element of the block?
 - Should we continue merging all props or should we limit ourselves to the ones impacted by the hook itself (style and classname): the idea here is I'm trying to think long term and see what's best in terms of limitting the API surface.

Reply:
Not sure about the second thing, but we should definitely make it clear that this hook is intended to be used on the root element of a block. Even if the hook throws errors when used improperly, we ought to help developers avoid that mistake in the first place.

Reply:
Sure, we can state that it should be used on the outer element.

We don't know what the future will be like, so I think we should continue merging props?

Reply:
> We don't know what the future will be like, so I think we should continue merging props?

to clarify, I don't know and I don't have a precise opinion. I can see how being merging everything is good in case we want to add something new to the block wrapper later but also wonder why people just don't add extra props manually. Let's leave it that way.

Reply:
Shall I just log an error when it's not a child of a block list?

Reply:
Tbh, I'm not sure if it should be enforced. I added a sentence for now that it should be added to the outer element.

Reply:
Just noting that checking whether the hook is being used properly may be a bit difficult due to the way "block alignments" are implemented for most blocks. (They use a wrapper around the block, with most block props being kept on the actual block, but the `wp-block` class being moved to the alignment wrapper.)